decompiler/config/jak2/all-types.gc

```diff
@@ -25472,37 +25618,35 @@
    (backup-nav-radius       float                    :offset-assert 44)
    (cam-side-shift          float                    :offset-assert 48)
    (enable-cam-side-shift   basic                    :offset-assert 52)
-   (gun?                    basic                    :offset-assert 56)
-   (controls                vehicle-controls :inline :offset-assert 60)
-   (accel-array             vector           :inline :offset-assert 80)
-   (local-accel             vector           :inline :offset-assert 208)
-   (pilot-trans             vector           :inline :offset-assert 224)
-   (pilot-quat              vector           :inline :offset-assert 240)
-   (pilot-scale             vector           :inline :offset-assert 256)
-   (pilot-time              time-frame                   :offset-assert 272)
-   (as-daxter?              basic                    :offset-assert 280)
-   (art-group-backup        basic                    :offset-assert 284)
+   (gun?                    symbol                   :offset-assert 56)
+   (controls                vehicle-controls         :offset 60)
+   (accel-array             vector           :inline :offset 80)
+   (local-accel             vector           :inline :offset 208)
+   (pilot-trans             vector           :inline :offset 224)
+   (pilot-quat              vector           :inline :offset 240)
+   (pilot-scale             vector           :inline :offset 256)
+   (pilot-time              time-frame               :offset 272)
+   (as-daxter?              basic                    :offset 280)
+   (art-group-backup        basic                    :offset 284)
    )
   :method-count-assert 9
   :size-assert         #x120
   :flag-assert         #x900000120
   )
-|#
 
 
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 ;; gun-h                          ;;
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 
-#|
 (deftype gun (process-drawable)
-  ((control       basic     :offset-assert 124)
-   (shadow-backup basic     :offset-assert 204)
-   (read-scale    basic     :offset-assert 208)
-   (gun-type      int32     :offset-assert 212)
-   (barrel        basic     :offset-assert 216)
-   (mag           UNKNOWN 4 :offset-assert 220)
-   (mag-scale     UNKNOWN 4 :offset-assert 236)
+  ((control       basic           :offset 128)
+   (shadow-backup basic           :offset 208)
+   (read-scale    basic           :offset 212)
```

Comment:
I think this field and the ones below it can be `:offset-assert`.

As you probably noticed, there's a mistake in all children of `process` where the fields are wrong by 4 bytes.